yt/visualization/plot_modifications.py

```diff
@@ -77,8 +77,11 @@ def _project_coords(self, plot, coord):
         """
         if len(coord) == 3:
             if not isinstance(coord, YTArray):
-                coord = plot.data.ds.arr(coord, "code_length")
-            coord.convert_to_units("code_length")
+                coord_copy = plot.data.ds.arr(coord, "code_length")
+            # coord is being copied so that if the user has a unyt_array already
+            # we don't change the user's version
+            else:
```

Comment:
Can you move the comment so that it's clearer what branch is discussed in it please ? After that I think I'll be able to approve + merge (unless CI fails)
```suggestion
            else:
                # coord is being copied so that if the user has a unyt_array already
                # we don't change the user's version
```